nncf/openvino/quantization/quantize_model.py

```diff
@@ -82,17 +111,17 @@ def native_quantize_if_op_impl(
         ignored_scope=ignored_scope,
         advanced_parameters=advanced_parameters,
     )
-
-    graph = GraphConverter.create_nncf_graph(model)
-    warning_model_no_batchwise_support(graph, advanced_parameters, model_type, OPERATIONS_OUTPUT_HAS_NO_BATCH_AXIS)
+    for graph in graphs.values():
+        if is_model_no_batchwise_support(graph, advanced_parameters, model_type, OPERATIONS_OUTPUT_HAS_NO_BATCH_AXIS):
+            nncf_logger.warning(BATCHWISE_STATISTICS_WARNING)
+            break
     if_ops_number = get_number_if_op(model)
-    all_models_number = if_ops_number * 2 + 1
     nncf_logger.info(
         f"The model consists of {if_ops_number} If node(-s) with then and else bodies. \
             Main model and all If bodies will be quantized recursively."
     )
     quantized_model, _ = apply_algorithm_if_bodies(
-        quantization_algorithm, model, graph, calibration_dataset, subset_size, 1, all_models_number
+        quantization_algorithm, model, graphs, calibration_dataset, subset_size, 1
```

Comment:
The basic idea of ​​quantization of model with if operation is that we divide the model into several models and process each one separately. My understanding is that you find if operations, capture all models and build nncf graph for each one before calling `apply_algorithm_if_bodies`. Thus could you feed all these information to apply_algorithm_if_bodies?